resources/resource_utils.py

```diff
@@ -0,0 +1,101 @@
+from typing import Optional, Dict, Any
+import hail as hl
+
+
+# Resource classes
+class BaseResource:
+    """
+    Generic abstract resource class.
+    """
+
+
+    def __init__(self, path: str, source_path: Optional[str] = None, attributes: Dict[str, Any] = None):
```

Comment:
Is `attributes` expected to be mostly used for resources that are imported from multiple files? Could it be named something more specific, like `other_sources`?